Automattic/wp-calypso
# Inline Help: Show first checklist task in inline help

As part of the new onboarding flow, we plan to send users to the Site Preview, and use the Inline Help pop-up to direct them to their first checklist task.

#### Changes proposed in this Pull Request

* Create a new `viewMode` for the wpcom-checklist called `ChecklistPrompt` which uses `ChecklistPromptTask` to display the first incomplete checklist task with title, description, estimated time, and a button to complete it.
* Adds `ChecklistPrompt` to inline help as a secondary view, only shown when a unique query string is present (currently `checklist`). We can use this query string to show/hide the prompt from any Calypso URL, and remove the query to dismiss the prompt.
* Account for new `viewMode` when tracking checklist task start in Tracks.
* Also see related PR #29068, which opens the Inline Help pop up with a special query string. Sending folks to `/view/?help&checklist` from signup/checkout will have the desired effect.

<img width="353" alt="screen shot 2018-12-13 at 12 17 08 pm" src="https://user-images.githubusercontent.com/2124984/49955544-12b0d980-fed1-11e8-884c-e83c15af2df8.png">

#### Testing instructions

* Switch to this PR
* Navigate to `/view` on a site that's eligible for the dotcom checklist.
* Add the query string to the URL: `?checklist`
* Open Inline Help and you should see the checklist prompt: first task title, description, and a button, as well as the checklist navigation below it.
